packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -0,0 +1,169 @@
+   */
+  @Prop() buttonText!: string;
+
+  /**
+   * Set the input to required and render the (Required) text.
+   */
+  @Prop() required?: boolean = false;
+
+  /**
+   * Allow the input to accept multiple files.
+   */
+  @Prop() multiple?: boolean = false;
+
+  /**
+   * A comma-separated list of unique file type specifiers.
+   */
+  @Prop() accept?: string;
+
+  /**
+   * The error message to render.
+   */
+  @Prop() error?: string;
+
+  /**
+   * Emit component-library-analytics events on the file input change event.
+   */
+   @Prop() enableAnalytics?: boolean = false;
+
+  /**
+   * The event emitted when the file input value changes.
+   */
+  @Event() vaChange: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when the
+   * file input changes and enableAnalytics is true.
+   */
+   @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private handleChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    this.vaChange.emit({files: target.files});
+    /**
+     * Clear the original input, otherwise events will be triggered
+     * with empty file arrays and sometimes uploading a file twice will
+     * not work.
+     */
+    target.value = null;
+
+    // TODO: Analytics details.
+    if (this.enableAnalytics) {
+      this.componentLibraryAnalytics.emit({
+        componentName: 'va-file-input',
+        action: 'change',
+        details: {
+          label: this.label
+        },
+      });
+    }
+  };
+
+  private handleButtonClick = () => { 
+    this.el.shadowRoot.getElementById('fileInputField').click();
+  }
+
+  connectedCallback() {
+    i18next.on('languageChanged', () => {
+      forceUpdate(this.el);
+    });
+  }
+
+  disconnectedCallback() {
+    i18next.off('languageChanged');
+  }
+
+  render() {
+    const { 
+      label,
+      name, 
+      buttonText, 
+      required, 
+      multiple,
+      accept,
+      error,
+    } = this;
+    return (
+      <Host>
+        {label && (
+          <label htmlFor="fileInputButton">
+            {label}
+            {required && <span class="required">{i18next.t('required')}</span>}
+          </label>
+        )}
+        <slot></slot>
+        <span id="error-message" role="alert">
+          {error && (
+            <Fragment>
+              <span class="sr-only">{i18next.t('error')}</span> 
+              {error}
+            </Fragment>
+          )}
+        </span>
+        <va-button
+          id="fileInputButton"
+          aria-label={i18next.t(label)}
+          onClick={this.handleButtonClick}
+          secondary
+          text={buttonText ? i18next.t(buttonText) : i18next.t('Upload file')}
+          aria-describedby={error ? 'error-message' : undefined}
+        />
+        <input
+            id="fileInputField"
+            multiple={multiple}
+            style={{ display: 'none' }}
```

Comment:
Could we have these styles live in `va-file-input.css`?

Reply:
@andresriveratoro That's a good question. I wasn't completely sure myself. The React version had it inline so I was just pulling it over here. I leaned towards keeping it because I felt like it made it a little more readable for what was going but I'm happy to move it into the styles though and just have a `.hidden` class.